src/main/scala/uk/gov/hmrc/hmrcfrontend/views/viewmodels/layout/CustomLayout.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hmrcfrontend.views.viewmodels.layout
+
+import play.twirl.api.Html
+import uk.gov.hmrc.govukfrontend.views.Aliases.PageLayout
+import uk.gov.hmrc.hmrcfrontend.views.Aliases.Header
+
+case class CustomLayout(
+  additionalHeadBlock: Option[Html] = None,
+  additionalScriptsBlock: Option[Html] = None,
+  beforeContentBlock: Option[Html] = None,
+  mainContentLayout: Option[Html => Html] = None,
+  pageLayout: Option[PageLayout => Html] = None,
+  headerContainerClasses: String = Header.defaultObject.containerClasses
+)
```

Comment:
@ellamdav rather than CustomLayout, what do you think about something like LayoutOverrides or TemplateOverrides? Initially I read it as "no main content layout" and "no page layout" by default, rather than you're not overriding the default (where the parameters before had the default values in the case class) so the name of the params are the same but the semantics of it's value are now different

Reply:
Yeah, I'm not really happy with how this bit now works.  The defaults were previously set using injected templates, but we can't do that in a simple case class constructor, so I've ended up adding that awkward `CustomLayout.noMainContentLayout`.

Reply:
renamed `CustomLayout` to `TemplateOverrides`